torch/_functorch/aot_autograd.py

```diff
@@ -2777,6 +2777,12 @@ def aot_dispatch_autograd(flat_fn, flat_args: List[Any], aot_config: AOTConfig,
                 # We are not clearing flat_args here because
                 # 1) There is a check in the the debug compiler at the end
                 # 2) It does not matter as these are fake tensors
+
+            # the compiler need to use this field to find the original modol outputs
+            # from the AOTAutograd fwd module's outputs. Thus compiler can make sure
+            # optimizations like layout optimization does not change those tensors'
+            # layout.
+            fw_module.meta["original_output_start_index"] = fw_metadata.num_mutated_inputs
```

Comment:
I feel like there should be a better way of passing this to backend compilers. Any thoughts @bdhirsh ?

Reply:
yea, this does not look pretty :( But I don't want to add it as an extra argument for the compiler function as well.

Reply:
@eellison currently has a PR up to expose all of `fw_metadata` directly in the global tracing context, that way inductor can get at it anywhere: https://github.com/pytorch/pytorch/pull/100652/files#r1212002707

I don't think this is great either, but we sort of agreed (?) that adding it to global state is less hairy than either stashing more random metadata on the fx module, or continually making changes to the backend calling convention.

Reply:
I added a TODO to access fw_metadata from the global tracing context after elias's PR is in.